numpy/core/_add_newdocs.py

```diff
@@ -2507,7 +2507,7 @@
 
 
 add_newdoc('numpy.core.multiarray', 'ndarray', ('__array__',
-    """ a.__array__(|dtype) -> reference if type unchanged, copy otherwise.
+    """ a.__array__(dtype=None) -> reference if type unchanged, copy otherwise.
```

Comment:
```suggestion
    """ a.__array__([dtype], /) -> reference if type unchanged, copy otherwise.
```

Passing `dtype=None` to `ndarray.__array__` is not the same as not passing it at all.

Also `/` there since the argument is positional only.

Reply:
```
In [2]: a = np.array([1, 2, 3])

In [4]: a.__array__()
Out[4]: array([1, 2, 3])

In [5]: a.__array__(dtype=None)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-5eab9c16cb1d> in <module>
----> 1 a.__array__(dtype=None)

TypeError: __array__() takes no keyword arguments

In [6]: a.__array__(None)
Out[6]: array([1., 2., 3.])
```